src/main/kotlin/com/ajaxproject/warehouse/service/WaybillServiceImpl.kt

```diff
@@ -1,130 +1,132 @@
 @Suppress("TooManyFunctions")
 class WaybillServiceImpl(
-    val waybillRepository: WaybillRepository,
-    val customerRepository: CustomerRepository,
-    val waybillProductRepository: WaybillProductRepository,
-    val productRepository: ProductRepository
+    val mongoWaybillRepository: MongoWaybillRepository,
+    val mongoProductRepository: MongoProductRepository,
+    val mongoCustomerRepository: MongoCustomerRepository
 ) : WaybillService {
 
-    override fun findAll(): List<WaybillDataLiteDto> {
-        return waybillRepository.findAll().map { it.mapToLiteDto() }
+    override fun findAllWaybills(): List<WaybillDataLiteDto> =
+        mongoWaybillRepository.findAll().map { it.mapToLiteDto() }
+
+    override fun getById(id: String): WaybillDataDto {
+        val waybill: MongoWaybill = mongoWaybillRepository.findById(ObjectId(id))
+            ?: throw NotFoundException("Waybill with id $id not found")
+        return waybill.mapToDataDto()
     }
 
-    override fun findById(id: Int): WaybillDataDto {
-        return waybillRepository.findById(id)
-            .orElseThrow { NotFoundException("Waybill with id $id not found") }
-            .mapToDataDto()
+    @Transactional
+    override fun updateWaybillInfo(infoUpdateDto: WaybillInfoUpdateDto, id: String): WaybillDataDto {
+        val mongoWaybill: MongoWaybill = mongoWaybillRepository.findById(ObjectId(id))
+            ?: throw NotFoundException("Waybill with id $id not found")
+        mongoCustomerRepository.findById(ObjectId(infoUpdateDto.customerId))
+            ?: throw NotFoundException("Customer with id ${infoUpdateDto.customerId} not found")
+        val updatedWaybill = mongoWaybill.setUpdatedData(infoUpdateDto)
+        return mongoWaybillRepository.save(updatedWaybill).mapToDataDto()
     }
 
+    @Transactional
     override fun createWaybill(createDto: WaybillCreateDto): WaybillDataDto {
-        val waybill = createDto.mapToEntity()
-        if (createDto.products == null) {
-            return waybillRepository.save(waybill).mapToDataDto()
-        }
         val errorList: MutableList<String> = mutableListOf()
-        val products: MutableList<Pair<Product, Int>> = mutableListOf()
-        for (productDto in createDto.products) {
-            productRepository.findById(productDto.id as Int).ifPresentOrElse(
-                { products.add(it to productDto.amount as Int) },
-                { errorList.add("Product with id ${productDto.id} not found") }
-            )
-        }
-        if (errorList.isNotEmpty()) {
-            throw NotFoundException(errorList)
-        }
-        waybillRepository.save(waybill)
-        products.forEach { (product, amount) ->
-            waybillProductRepository.saveProductAndChangeAmount(waybill, product, amount)
-        }
-        return waybill.mapToDataDto()
+        mongoCustomerRepository.findById(ObjectId(createDto.customerId))
+            ?: errorList.add("Customer with id ${createDto.customerId} not found")
```

Comment:
```suggestion
            ?: listOf("Customer with id ${createDto.customerId} not found")
```

Reply:
Why not to add error message to existing list? I am using this list later and also I cannot use it like 
    
```kotlin
val errorList: MutableList<String> = mongoCustomerRepository.findById(ObjectId(createDto.customerId))
            ?: mutableListOf("Customer with id ${createDto.customerId} not found")
``` 

because of different return types